crates/gitbutler-stack/src/stack_context.rs

```diff
@@ -0,0 +1,33 @@
+use anyhow::Result;
+use gitbutler_command_context::CommandContext;
+
+use crate::{Target, VirtualBranchesHandle};
+
+pub struct StackContext<'repositroy> {
+    repository: &'repositroy git2::Repository,
+    target: Target,
+}
+
+impl<'repository> StackContext<'repository> {
+    pub fn repository(&self) -> &'repository git2::Repository {
+        self.repository
+    }
+
+    pub fn target(&self) -> &Target {
+        &self.target
+    }
+}
+
+impl<'repository> TryFrom<&'repository CommandContext> for StackContext<'repository> {
```

Comment:
```
into_…(self) -> NewThing
to_...(&self) -> NewThing
as_…(&self) -> &PartOfSelf
```
Thank you for sharing this! I found it educational for future code that I write!

---
PS: specifically on this issue, we chatted about it on Discord about just passing the 2 dependencies instead of making a new context https://discord.com/channels/1060193121130000425/1272848546399387669/1305972659926073456

Reply:
As an update, we've decided to keep the `StackContext` for now, and will come back to the extra refactor later